### ISSUE TITLE
No recording happening with a lot of topics

### ISSUE THREAD

**Issue by dbruggner** [ros2/rosbag2#1062]
## Description
When recording a lot of topics rosbag2 doesn't subscribe to all topics and data is not recorded.

### Expected Behavior
It is expected that rosbag2 will subscribe to all topics (even if these are a lot) and record all the data.

### Actual Behavior
This behavior can be reproduced with the following launch file. For example rosbag2 records fine with 10 `image_publishers` but not with 100 `image_publishers`. The log shows the output of `ros2 bag info` for a bag file recorded for 100 `image_publishers`
Launch file:
```
import launch
from launch import LaunchDescription
from launch_ros.actions import Node
from launch.actions import ExecuteProcess
from launch.substitutions import LaunchConfiguration

from typing import Final, List
from datetime import datetime

NAMESPACE: Final[str] = "/my_space"

def generate_launch_description():

    nodes: Final[List[Node]] = []

    for i in range(100):
        nodes.append(
            Node(
                name="image_pub_" + str(i),
                executable="image_publisher_node",
                package="image_publisher",
                namespace=NAMESPACE + "/_" + str(i),
                parameters=[
                    {"publish_rate" : 20.0},
                    {"filename" : "/home/user/Pictures/Test.jpg"},
                    ],
                ))
    
    bag_record: Final[List[ExecuteProcess]] = [
        ExecuteProcess(
            cmd=["ros2", "bag", "record",  "-a", "--include-hidden-topics", "-o","/tmp/"+datetime.now().strftime("%Y%m%d_%H%M%S") ],
            output="screen",
            log_cmd=True,
        ),
    ]

    ld = LaunchDescription()

    ld.add_action(LaunchDescription(nodes))
    ld.add_action(LaunchDescription(bag_record))

    return ld
```

Log for 100 `image_publishers`:
```
Files:             20220809_093334_0.db3
Bag size:          889.9 KiB
Storage id:        sqlite3
Duration:          34.364s
Start:             Aug  9 2022 09:33:51.367 (1660037631.367)
End:               Aug  9 2022 09:34:25.732 (1660037665.732)
Messages:          1785
Topic information: Topic: /nps/_59/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 18 | Serialization Format: cdr
                   Topic: /nps/_61/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 11 | Serialization Format: cdr
                   Topic: /nps/_28/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_13/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 5 | Serialization Format: cdr
                   Topic: /nps/_70/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_32/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_56/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_38/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 1 | Serialization Format: cdr
                   Topic: /nps/_87/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 5 | Serialization Format: cdr
                   Topic: /nps/_49/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_10/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_96/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_88/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_53/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 18 | Serialization Format: cdr
                   Topic: /nps/_64/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_53/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_27/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 7 | Serialization Format: cdr
                   Topic: /nps/_37/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_81/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 19 | Serialization Format: cdr
                   Topic: /nps/_46/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 7 | Serialization Format: cdr
                   Topic: /nps/_6/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 6 | Serialization Format: cdr
                   Topic: /nps/_80/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_87/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_65/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_80/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 14 | Serialization Format: cdr
                   Topic: /nps/_58/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 19 | Serialization Format: cdr
                   Topic: /nps/_88/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 8 | Serialization Format: cdr
                   Topic: /nps/_55/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 3 | Serialization Format: cdr
                   Topic: /nps/_13/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_94/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_3/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_89/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 2 | Serialization Format: cdr
                   Topic: /nps/_71/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_90/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_28/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 7 | Serialization Format: cdr
                   Topic: /nps/_11/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 15 | Serialization Format: cdr
                   Topic: /nps/_7/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 10 | Serialization Format: cdr
                   Topic: /nps/_96/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 8 | Serialization Format: cdr
                   Topic: /nps/_35/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 26 | Serialization Format: cdr
                   Topic: /nps/_81/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_98/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_74/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 13 | Serialization Format: cdr
                   Topic: /nps/_35/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_75/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 11 | Serialization Format: cdr
                   Topic: /nps/_4/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 14 | Serialization Format: cdr
                   Topic: /nps/_32/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 5 | Serialization Format: cdr
                   Topic: /nps/_83/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 21 | Serialization Format: cdr
                   Topic: /nps/_76/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 10 | Serialization Format: cdr
                   Topic: /nps/_40/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_86/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 32 | Serialization Format: cdr
                   Topic: /nps/_47/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_5/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 24 | Serialization Format: cdr
                   Topic: /nps/_48/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_62/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 7 | Serialization Format: cdr
                   Topic: /nps/_14/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 7 | Serialization Format: cdr
                   Topic: /parameter_events | Type: rcl_interfaces/msg/ParameterEvent | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_54/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 25 | Serialization Format: cdr
                   Topic: /nps/_46/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_95/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_3/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 5 | Serialization Format: cdr
                   Topic: /nps/_47/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 16 | Serialization Format: cdr
                   Topic: /nps/_97/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_64/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 15 | Serialization Format: cdr
                   Topic: /nps/_45/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 9 | Serialization Format: cdr
                   Topic: /nps/_12/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 13 | Serialization Format: cdr
                   Topic: /nps/_31/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_76/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_42/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 29 | Serialization Format: cdr
                   Topic: /nps/_0/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_2/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_78/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 10 | Serialization Format: cdr
                   Topic: /nps/_9/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 16 | Serialization Format: cdr
                   Topic: /nps/_79/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 10 | Serialization Format: cdr
                   Topic: /nps/_9/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_74/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_99/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 25 | Serialization Format: cdr
                   Topic: /nps/_71/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 5 | Serialization Format: cdr
                   Topic: /nps/_85/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_93/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 21 | Serialization Format: cdr
                   Topic: /nps/_22/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_65/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 10 | Serialization Format: cdr
                   Topic: /nps/_75/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_90/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 19 | Serialization Format: cdr
                   Topic: /nps/_39/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 23 | Serialization Format: cdr
                   Topic: /nps/_95/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 7 | Serialization Format: cdr
                   Topic: /nps/_24/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 43 | Serialization Format: cdr
                   Topic: /nps/_73/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 52 | Serialization Format: cdr
                   Topic: /nps/_36/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_25/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 24 | Serialization Format: cdr
                   Topic: /nps/_40/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 11 | Serialization Format: cdr
                   Topic: /nps/_50/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 30 | Serialization Format: cdr
                   Topic: /nps/_69/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 6 | Serialization Format: cdr
                   Topic: /nps/_2/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 68 | Serialization Format: cdr
                   Topic: /nps/_69/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_20/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 12 | Serialization Format: cdr
                   Topic: /nps/_19/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_63/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 22 | Serialization Format: cdr
                   Topic: /nps/_63/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_48/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 4 | Serialization Format: cdr
                   Topic: /nps/_52/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 10 | Serialization Format: cdr
                   Topic: /nps/_91/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_99/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_70/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 37 | Serialization Format: cdr
                   Topic: /nps/_92/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 21 | Serialization Format: cdr
                   Topic: /nps/_30/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_15/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_30/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 18 | Serialization Format: cdr
                   Topic: /nps/_78/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_60/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 16 | Serialization Format: cdr
                   Topic: /nps/_8/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 28 | Serialization Format: cdr
                   Topic: /nps/_72/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 5 | Serialization Format: cdr
                   Topic: /nps/_92/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_34/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 14 | Serialization Format: cdr
                   Topic: /nps/_77/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_51/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_45/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_8/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_16/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_7/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_21/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_68/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_73/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_42/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_86/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_50/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_61/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_85/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 29 | Serialization Format: cdr
                   Topic: /nps/_1/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_11/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_66/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_94/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 19 | Serialization Format: cdr
                   Topic: /nps/_79/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_77/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 8 | Serialization Format: cdr
                   Topic: /nps/_52/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_26/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_33/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_23/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_66/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 25 | Serialization Format: cdr
                   Topic: /nps/_82/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 14 | Serialization Format: cdr
                   Topic: /nps/_0/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 20 | Serialization Format: cdr
                   Topic: /nps/_19/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 9 | Serialization Format: cdr
                   Topic: /nps/_43/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 22 | Serialization Format: cdr
                   Topic: /nps/_18/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 25 | Serialization Format: cdr
                   Topic: /nps/_91/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 19 | Serialization Format: cdr
                   Topic: /nps/_24/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_33/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 11 | Serialization Format: cdr
                   Topic: /nps/_1/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 26 | Serialization Format: cdr
                   Topic: /nps/_60/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_97/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 11 | Serialization Format: cdr
                   Topic: /nps/_12/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_58/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_26/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 22 | Serialization Format: cdr
                   Topic: /nps/_25/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_34/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_93/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_29/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_29/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 52 | Serialization Format: cdr
                   Topic: /nps/_51/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 9 | Serialization Format: cdr
                   Topic: /nps/_22/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 8 | Serialization Format: cdr
                   Topic: /nps/_27/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_54/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_67/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_62/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_23/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 2 | Serialization Format: cdr
                   Topic: /nps/_43/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_44/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_67/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 16 | Serialization Format: cdr
                   Topic: /nps/_20/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_15/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 7 | Serialization Format: cdr
                   Topic: /nps/_59/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_16/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 30 | Serialization Format: cdr
                   Topic: /nps/_56/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 24 | Serialization Format: cdr
                   Topic: /nps/_49/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 24 | Serialization Format: cdr
                   Topic: /nps/_41/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_38/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_82/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_37/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 19 | Serialization Format: cdr
                   Topic: /nps/_36/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 12 | Serialization Format: cdr
                   Topic: /nps/_6/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /rosout | Type: rcl_interfaces/msg/Log | Count: 206 | Serialization Format: cdr
                   Topic: /nps/_39/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_4/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_21/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 24 | Serialization Format: cdr
                   Topic: /nps/_98/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 3 | Serialization Format: cdr
                   Topic: /nps/_17/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_10/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 5 | Serialization Format: cdr
                   Topic: /nps/_17/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 11 | Serialization Format: cdr
                   Topic: /nps/_31/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 10 | Serialization Format: cdr
                   Topic: /nps/_55/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_14/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_41/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 4 | Serialization Format: cdr
                   Topic: /nps/_68/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 20 | Serialization Format: cdr
                   Topic: /nps/_89/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_18/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_5/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_44/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 7 | Serialization Format: cdr
                   Topic: /nps/_72/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
                   Topic: /nps/_83/image_raw | Type: sensor_msgs/msg/Image | Count: 0 | Serialization Format: cdr
```

Log for 10 `image_publishers`:
```
Files:             20220809_093458_0.db3
Bag size:          5.6 GiB
Storage id:        sqlite3
Duration:          12.170s
Start:             Aug  9 2022 09:34:58.598 (1660037698.598)
End:               Aug  9 2022 09:35:10.769 (1660037710.769)
Messages:          3375
Topic information: Topic: /nps/_5/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 238 | Serialization Format: cdr
                   Topic: /nps/_4/image_raw | Type: sensor_msgs/msg/Image | Count: 93 | Serialization Format: cdr
                   Topic: /nps/_2/image_raw | Type: sensor_msgs/msg/Image | Count: 78 | Serialization Format: cdr
                   Topic: /nps/_2/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 243 | Serialization Format: cdr
                   Topic: /nps/_0/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 243 | Serialization Format: cdr
                   Topic: /nps/_4/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 238 | Serialization Format: cdr
                   Topic: /nps/_6/image_raw | Type: sensor_msgs/msg/Image | Count: 17 | Serialization Format: cdr
                   Topic: /nps/_9/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 237 | Serialization Format: cdr
                   Topic: /parameter_events | Type: rcl_interfaces/msg/ParameterEvent | Count: 0 | Serialization Format: cdr
                   Topic: /rosout | Type: rcl_interfaces/msg/Log | Count: 28 | Serialization Format: cdr
                   Topic: /nps/_3/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 237 | Serialization Format: cdr
                   Topic: /nps/_1/image_raw | Type: sensor_msgs/msg/Image | Count: 46 | Serialization Format: cdr
                   Topic: /nps/_7/image_raw | Type: sensor_msgs/msg/Image | Count: 94 | Serialization Format: cdr
                   Topic: /nps/_9/image_raw | Type: sensor_msgs/msg/Image | Count: 144 | Serialization Format: cdr
                   Topic: /nps/_0/image_raw | Type: sensor_msgs/msg/Image | Count: 185 | Serialization Format: cdr
                   Topic: /nps/_1/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 242 | Serialization Format: cdr
                   Topic: /nps/_6/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 233 | Serialization Format: cdr
                   Topic: /nps/_5/image_raw | Type: sensor_msgs/msg/Image | Count: 96 | Serialization Format: cdr
                   Topic: /nps/_3/image_raw | Type: sensor_msgs/msg/Image | Count: 87 | Serialization Format: cdr
                   Topic: /nps/_8/image_raw | Type: sensor_msgs/msg/Image | Count: 120 | Serialization Format: cdr
                   Topic: /nps/_8/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 241 | Serialization Format: cdr
                   Topic: /nps/_7/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 235 | Serialization Format: cdr
```

## To Reproduce
1. Install `ros-galactic-image-publisher`
2. Create a ROS package with the given launch file
3. `source your_workspace/setup.bash`
4. Run the launch file with `ros2 launch your_package the_launch_file.py
5. `ros2 bag info /tmp/2022*`

## System (please complete the following information)
 - OS: Ubuntu Focal
 - ROS 2 Distro: Galactic
 - Version: galactic

**Comment by MichaelOrlov** [comment]
@dbruggner What is the size of the publishing image? 
And what DDS vendor (`FastRTPS`, `CycloneDDS` or `RTI Connext DDS`) did you use?

**Comment by dbruggner** [comment]
The size of the publishing image can be anything. I tested also with 1 pixel sized images.
The used DDS is Cyclone.

**Comment by dbruggner** [comment]
It seems that `get_topic_names_and_types` https://github.com/ros2/rosbag2/blob/rolling/rosbag2_transport/src/rosbag2_transport/recorder.cpp#L262 returns `0` most of the times when a lot of topics are available. It happens also when I want to record only one topic.

**Comment by MichaelOrlov** [comment]
@dbruggner We have came across with the same issue some time ago in Apex.OS.  
After comprehensive analysis we found out that root cause of the problem not in the `rosbag2` itself rather in CycloneDDS.

`get_topic_names_and_types()` will only return max 100 topics, because we are using graph info inside where we read the samples (and not take), the CycloneDDS read API sets the max samples to 100 internally [here](https://github.com/eclipse-cyclonedds/cyclonedds/blob/c1f7ed4c2bd14542413b24406072c2a8b903f2b0/src/core/ddsc/src/dds_read.c#L222) which causes this issue.
